src/mavsdk/plugins/telemetry/telemetry_impl.cpp

```diff
@@ -1176,6 +1181,42 @@ void TelemetryImpl::process_battery_status(const mavlink_message_t& message)
     }
 }
 
+void TelemetryImpl::process_battery_status_v2(const mavlink_message_t& message)
+{
+    mavlink_battery_status_v2_t bat_status;
+    mavlink_msg_battery_status_v2_decode(&message, &bat_status);
+
+    _has_bat_status = true;
+
+    Telemetry::Battery new_battery;
+    new_battery.id = bat_status.id;
+    new_battery.voltage_v = bat_status.voltage; 
+    //Is it correct to set this to NaN for UINT32_MAX? HOw do you specify = NaN?
```

Comment:
@julianoes The old version seems to ignore possibility that these values might be arriving as UINT32_MAX or whatever. 
The [doc says](https://mavsdk.mavlink.io/main/en/cpp/api_reference/structmavsdk_1_1_telemetry_1_1_battery.html#structmavsdk_1_1_telemetry_1_1_battery_1aefd1d5569ff2a69bf507b422cbe75742)
> `float mavsdk::Telemetry::Battery::voltage_v {float(NAN)}`

Does that mean we need to check for a UINT32_MAX and set to NaN? Hints?

Reply:
Yes, good point. We should check for it and set NAN if it's `std::limits<uint32_t>::max`, the C++ way or UINT32_MAX :joy:.

Reply:
Cool. voltage fixed above. But leaving battery remaining until I understand if we can/should make that an int